Aquarius/Src/Aquarius/Events/MouseEvent.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include "Event.h"
+
+#include <GLFW/glfw3.h>
+
+
+namespace Aquarius {
+
+    class MouseMovedEvent : public Event
+    {
+    public:
+        MouseMovedEvent(float xCoordinate, float yCoordinate)
+                : m_XCoordinate(xCoordinate), m_YCoordinate(yCoordinate) {}
+
+        EVENT_TYPE(MouseMovedEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("MouseMovedEvent: %v, %v", m_XCoordinate, m_YCoordinate);
+        }
+
+        float getXCoordinate()
+        {
+            return m_XCoordinate;
+        }
+
+        float getYCoordinate()
+        {
+            return m_YCoordinate;
+        }
+
+    private:
+        float m_XCoordinate;
+        float m_YCoordinate;
+    };
+
+    class MouseScrolledEvent : public Event
+    {
+    public:
+        MouseScrolledEvent(float xCoordinate, float yCoordinate)
+                : m_XCoordinate(xCoordinate), m_YCoordinate(yCoordinate) {}
+
+        EVENT_TYPE(MouseScrolledEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("MouseScrolledEvent: %v, %v", m_XCoordinate, m_YCoordinate);
+        }
+
+        float getXCoordinate()
+        {
+            return m_XCoordinate;
+        }
+
+        float getYCoordinate()
+        {
+            return m_YCoordinate;
+        }
+
+    private:
+        float m_XCoordinate;
+        float m_YCoordinate;
+    };
+
+    class MouseButtonEvent : public Event
+    {
+    public:
+        int getCode()
+        {
+            return m_buttonCode;
+        }
+
+    protected:
+        int m_buttonCode;
+
+        MouseButtonEvent(int buttonCode) : m_buttonCode(buttonCode) {}
+    };
+
+    class MouseButtonPressedEvent : public MouseButtonEvent
+    {
+    public:
+        MouseButtonPressedEvent(int buttonCode) : MouseButtonEvent(buttonCode) {}
+
+        EVENT_TYPE(MouseButtonPressedEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("MouseButtonPressedEvent: %v", m_buttonCode);
+        }
+    };
+
+    class MouseButtonReleasedEvent : public MouseButtonEvent
+    {
+    public:
+        MouseButtonReleasedEvent(int buttonCode) : MouseButtonEvent(buttonCode) {}
+
+        EVENT_TYPE(MouseButtonReleasedEvent)
+
+        void logEvent()
+        {
+            AQ_CORE_TRACE("MouseButtonReleasedEvent: %v", m_buttonCode);
+        }
+    };
+
+    class MouseInWindowEvent : public Event
+    {
+    public:
+        MouseInWindowEvent(int inWindow) : m_inWindow(inWindow) {}
+
+        EVENT_TYPE(MouseInWindowEvent)
+
+        void logEvent()
+        {
+            if (inWindow() == 1)
+            {
+                AQ_CORE_TRACE("Cursor is in the window.");
+            }
+
+            else
+            {
+                AQ_CORE_TRACE("Cursor is not in the window.");
+            }
+        }
+
+        int inWindow()
+        {
+            return m_inWindow;
+        }
+
+    private:
+        int m_inWindow;
```

Comment:
Should probably be a bool

Reply:
It has to be an int due to the glfw callback, Daniel mentioned this before and thought it was something to do with C. That being said I have updated the inWindow() method to return a bool value rather than an int which could be used to check if the cursor is in the window. I will have this updated in my commit this evening.

Reply:
I mean you could cast the int returned by the callback to a bool before creating this event right? Doesn't really matter though

Reply:
Nvm, did a cast.